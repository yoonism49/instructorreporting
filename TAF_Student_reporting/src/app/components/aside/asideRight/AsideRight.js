// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';


const AsideRight = ({
  children,
  isAnimated,
  isExpanded
}) => (
  <aside className={
    cx({
      'right-side':             true,
      'right-side--top-margin': true,
      'aside-right-animated':   isAnimated,
      'strech':                 isExpanded
    })}>
    { children }
  </aside>
);

AsideRight.propTypes = {
  children:   PropTypes.node.isRequired,
  isAnimated: PropTypes.bool.isRequired,
  isExpanded: PropTypes.bool.isRequired
};

AsideRight.defaultProps = {
  isAnimated: true,
  isExpanded: false
};

export default AsideRight;
