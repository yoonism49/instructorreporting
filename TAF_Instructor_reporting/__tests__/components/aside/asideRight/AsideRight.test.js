'use strict';

import React          from 'react';
import AsideRight     from '../../../../src/app/components/aside/asideRight/AsideRight';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme


describe('AsideRight component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <AsideRight
          isAnimated
          isExpanded>
          <p>
            Aside right children
          </p>
        </AsideRight>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
