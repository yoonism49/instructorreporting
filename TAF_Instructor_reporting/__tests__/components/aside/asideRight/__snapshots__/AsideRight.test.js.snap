// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideRight component renders as expected 1`] = `
<div>
  <aside
    className="right-side right-side--top-margin aside-right-animated strech"
  >
    <p>
      Aside right children
    </p>
  </aside>
</div>
`;
